optuna/integration/mlflow.py

```diff
@@ -155,15 +158,19 @@ def __call__(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial) -
             tags=self._mlflow_kwargs.get("tags"),
         ):
 
-            # This sets the metrics for MLflow.
-            self._log_metrics(trial.values)
-
             # This sets the params for MLflow.
             self._log_params(trial.params)
 
             # This sets the tags for MLflow.
             self._set_tags(trial, study)
 
+            if trial.state == optuna.trial.TrialState.FAIL:
+                # Values are set to None when trial fails
+                mlflow.end_run(status="FAILED")
```

Comment:
We are within context manager provided by `mlflow.start_run` here, so this function will be called twice for the same run - here and again when exiting context manager. I might be wrong, but this could cause some weird behavior when there are more than one active runs (e.g. when using `track_in_mlflow` decorator). Can we add test cases to make sure existing functionality is not broken?